### ISSUE TITLE
Easier way to remove ctrl-w delay with vim mode

### ISSUE THREAD

**Issue by iluuu1994** [zed-industries/zed#13315]
### Check for existing issues

- [X] Completed

### Describe the feature

Vim-mode in zed declares a plethora of keymaps starting with `ctrl-w`. I personally don't use any of them, and just use `ctrl-w` to close the current tab, but want to remove the annoying 1s delay. However, getting this behavior is not quite straight-forward. I needed to add the following keymap:

<details><summary>Details</summary>
<p>

```json
[
  {
    "context": "Editor",
    "bindings": {
      "ctrl-w left": null,
      "ctrl-w right": null,
      "ctrl-w up": null,
      "ctrl-w down": null,
      "ctrl-w h": null,
      "ctrl-w l": null,
      "ctrl-w k": null,
      "ctrl-w j": null,
      "ctrl-w ctrl-h": null,
      "ctrl-w ctrl-l": null,
      "ctrl-w ctrl-k": null,
      "ctrl-w ctrl-j": null,
      "ctrl-w shift-left": null,
      "ctrl-w shift-right": null,
      "ctrl-w shift-up": null,
      "ctrl-w shift-down": null,
      "ctrl-w shift-h": null,
      "ctrl-w shift-l": null,
      "ctrl-w shift-k": null,
      "ctrl-w shift-j": null,
      "ctrl-w g t": null,
      "ctrl-w ctrl-g t": null,
      "ctrl-w g shift-t": null,
      "ctrl-w ctrl-g shift-t": null,
      "ctrl-w w": null,
      "ctrl-w ctrl-w": null,
      "ctrl-w p": null,
      "ctrl-w ctrl-p": null,
      "ctrl-w shift-w": null,
      "ctrl-w ctrl-shift-w": null,
      "ctrl-w v": null,
      "ctrl-w ctrl-v": null,
      "ctrl-w s": null,
      "ctrl-w shift-s": null,
      "ctrl-w ctrl-s": null,
      "ctrl-w c": null,
      "ctrl-w ctrl-c": null,
      "ctrl-w q": null,
      "ctrl-w ctrl-q": null,
      "ctrl-w o": null,
      "ctrl-w ctrl-o": null,
      "ctrl-w n": null,
      "ctrl-w ctrl-n": null,
      "ctrl-w d": null,
      "ctrl-w g d": null,
      "ctrl-w shift-d": null,
      "ctrl-w g shift-d": null,
      "ctrl-w space": null,
      "ctrl-w g space": null
    }
  }
]
``` 

</p>
</details> 

It's also fragile, because any new shortcut added to zed in the future will re-add the delay. It would be great if there was a better way to achieve this, e.g. through a wildcard of some kind:

```json
[
  {
    "context": "Editor",
    "bindings": {
      "ctrl-w *": null,
  }
]
```

If you don't think this needs better support, feel free to close this issue.

**Comment by sibouras** [comment]
+1, i use `ctrl-j/k` in vim mode to scroll and i had to unbind all the ctrl-k keymaps

**Comment by xzbdmw** [comment]
> +1, i use `ctrl-j/k` in vim mode to scroll and i had to unbind all the ctrl-k keymaps

You can try karabiner to map c-k to something else in zed to avoid unmapping all of them

**Comment by sibouras** [comment]
@xzbdmw good solution thanks

**Comment by 0x2CA** [comment]
If `nowait` is implemented it can do immediate processing for `<c-w>`
https://github.com/zed-industries/zed/issues/12539#issuecomment-2230183940

**Comment by xzbdmw** [comment]
Hmm, doesn’t nowait only applys to buffer local keymaps

**Comment by iluuu1994** [comment]
It seems my workaround broke. I can't remove the delay now. :slightly_frowning_face:

**Comment by ConradIrwin** [comment]
As of #14942, user-specified single key bindings should take precedence over built in multi-key bindings, so in theory just adding:

```
"bindings": {"ctrl-w": "pane::CloseActiveItem"}
```

should be sufficient. Please let me know if not

**Comment by iluuu1994** [comment]
@ConradIrwin To clarify, the issue described here is not that the custom keybinding didn't work. Adding `"ctrl-w": "pane::CloseActiveItem"` does work and seems to take precedence. However, there is a delay that cannot be removed. In the original description, I added a workaround that no longer seems to work. The keybinding triggers after 1s of no inputs, or when entering some other input.

**Comment by ConradIrwin** [comment]
Great to hear, and sorry for not closing this out sooner!

**Comment by StephanMoeller** [comment]
TLDR: adding the following will disable all built-ins and will remove any delay:

{
    "context": "Editor",
    "bindings": { "ctrl-w": "pane::CloseActiveItem" }
  }
